piksi_tools/console/settings_view.py

```diff
@@ -507,41 +492,76 @@ def display_ins_output_hint(self):
-                               "    In order for the \"Ins Output Mode\" setting to take effect, it is necessary to save the    \n" \
-                               "    current settings to device flash and then power cycle your device.    \n\n" \
-                               "    Choose \"OK\" to immediately save settings to device flash and send the software reset command.    \n" \
-                               "    The software reset will temporarily interrupt the console's connection to the device but it   \n" \
-                               "    will recover on its own.    \n\n"
+            "    In order for the \"Ins Output Mode\" setting to take effect, it is necessary to save the    \n" \
+            "    current settings to device flash and then power cycle your device.    \n\n" \
+            "    Choose \"OK\" to immediately save settings to device flash and send the software reset command.    \n" \
+            "    The software reset will temporarily interrupt the console's connection to the device but it   \n" \
+            "    will recover on its own.    \n\n"
 
         confirm_prompt2.run(block=False)
 
-    def _send_pending_settings_by_index(self):
-        for eachindex in self.pending_settings:
-            self.link(MsgSettingsReadByIndexReq(index=eachindex))
-
-    def _restart_retry_thread(self):
-        if self.retry_pending_read_index_thread:
-            self.retry_pending_read_index_thread.stop()
-        self.retry_pending_read_index_thread = TimedDelayStoppableThread(
-            SETTINGS_RETRY_TIMEOUT,
-            target=self._send_pending_settings_by_index, args=[])
-        self.retry_pending_read_index_thread.start()
-
-    def _settings_read_by_index(self):
-        self.enumindex = 0          # next index to ask for
-        self.pending_settings = []  # list of settings idices we've asked for
-        self.ordering_counter = 0   # helps make deterministic order of settings
-        self.setup_pending = True   # guards against receipt of multiple "done" msgs
-        # queue up BATCH_WINDOW settings indices to read
-        self.pending_settings = list(range(self.enumindex, self.enumindex + BATCH_WINDOW))
-        self.enumindex += BATCH_WINDOW
-        self._send_pending_settings_by_index()
-        # start a thread that will resend any read indexes that haven't come
-        self._restart_retry_thread()
+    def _settings_unconfirm_all(self):
+        # Clear the tabular editor
+        del self.settings_list[:]
+        for section in self.settings.keys():
+            for name in self.settings[section].keys():
+                self.settings[section][name].confirmed_set = False
+
+    def _settings_read_all(self):
+        self._settings_unconfirm_all()
+        self.workqueue.put(self._read_all_thread)
+
+    def _read_all_thread(self):
+        settings_list = self.settings_api.read_all()
+
+        idx = 0
+
+        for setting in settings_list:
+            section = setting['section']
```

Comment:
I presume that the libsettings side returns strings that work in both python 2 and 3... have you tested it with both 2.7 and 3.5?

Reply:
Doesn't seem to work under Python 3.5.0

```
pasi@pasi-linux:~/swiftnav/libsettings/python$ python settings_client.py --tcp -p 192.168.1.25:55555
('192.168.1.25', 55555)
Creating Settings
b'Building settings framework'
Traceback (most recent call last):
  File "settings_client.py", line 105, in <module>
    main()
  File "settings_client.py", line 90, in main
    print("solution.elevation_mask =", s.read("solution", "elevation_mask"))
  File "python/libsettings.pyx", line 160, in libsettings.Settings.read
    bytes(section),
TypeError: string argument without an encoding
pasi@pasi-linux:~/swiftnav/libsettings/python$ python --version
Python 3.5.0
```

Reply:
https://github.com/swift-nav/libsettings/pull/31